testsuite/python/observable_chain.py

```diff
@@ -181,5 +196,11 @@ def place_particles(bl, offset):
                         res_obs_chain, [dih1, dih2], decimal=9,
                         err_msg="Data did not agree for observable ParticleDihedrals")
 
+    def test_chain_angles(self):
+        self.system.part.clear()
+        self.system.part.add(pos= np.array([np.linspace(0, self.system.box_l[0], 20)] * 3).T + np.random.random((20, 3)))
```

Comment:
This test alters the state of the class. If this test starts before the two others, they could fail. From [unittest/Organizing test code](https://docs.python.org/3/library/unittest.html#organizing-test-code):
> The order in which the various tests will be run is determined by sorting the test method names with respect to the built-in ordering for strings.

This can happen if someone renames the method to `test_ChainAngles` for consistency with the other methods in this class. I tried it and it still passed, because the original test methods work on particle slices. I think it would be safer if you didn't create particles in your test, increased `n_parts` to 20 and only updated the positions of these `n_parts` particles. Alternatively, you could move the creation of particles from the `setUpClass()` to the individual test methods, and call `self.system.part.clear()` in a `tearDown()`.